service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateNetworkSecurityGroupStep.java

```diff
@@ -0,0 +1,85 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.landingzone.library.landingzones.deployment.LandingZoneTagKeys;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneResource;
+import bio.terra.landingzone.stairway.flight.LandingZoneFlightMapKeys;
+import bio.terra.landingzone.stairway.flight.ResourceNameProvider;
+import bio.terra.landingzone.stairway.flight.ResourceNameRequirements;
+import bio.terra.stairway.FlightContext;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateNetworkSecurityGroupStep extends BaseResourceCreateStep {
+  private static final Logger logger =
+      LoggerFactory.getLogger(CreateNetworkSecurityGroupStep.class);
+  public static final String NSG_ID = "NSG_ID";
+  public static final String NSG_RESOURCE_KEY = "NSG";
+
+  public CreateNetworkSecurityGroupStep(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameProvider resourceNameProvider) {
+    super(armManagers, parametersResolver, resourceNameProvider);
+  }
+
+  @Override
+  protected void createResource(FlightContext context, ArmManagers armManagers) {
+    var landingZoneId =
+        getParameterOrThrow(
+            context.getInputParameters(), LandingZoneFlightMapKeys.LANDING_ZONE_ID, UUID.class);
+    var nsgName = resourceNameProvider.getName(getResourceType());
+
+    var nsg =
```

Comment:
Is there a possibility of a conflict if this step is retried? Can that be handled?

Reply:
Fixed.